### DISCUSSION TITLE
Validate each forms with useFormState

### DISCUSSION THREAD

**Discussion by sabazusi** [react-hook-form/react-hook-form#8439] [Q&A]
Hi,

I am trying to monitor the status of each form using `useFormState`. However, when multiple forms are invalid at the same time, there is unexpected behavior.

**Codesandbox link**
https://codesandbox.io/s/quiet-flower-lwwsq0

**To Reproduce**
1. Remove value and make invalid any one of the forms.
2. Make invalid any one of the further forms.
3. 2nd form is not invalid, and do not change valid status.

||1|2|
|-|-|-|
|<img width="335" alt="スクリーンショット 2022-06-03 14 24 33" src="https://user-images.githubusercontent.com/962567/171791939-4f6c3549-3e61-457c-8de4-fb28f8f8ec93.png">|<img width="335" alt="スクリーンショット 2022-06-03 14 24 46" src="https://user-images.githubusercontent.com/962567/171791976-e37b6a15-e456-43b0-b741-6318a3c13104.png">|<img width="335" alt="スクリーンショット 2022-06-03 14 24 51" src="https://user-images.githubusercontent.com/962567/171792004-cf3ea55e-b4f1-4a21-8d01-7e8a52f64ba4.png">|

**Expected behavior**
2nd form status is invalid in step 2.

Any suggestions on how to accomplish this? Thank you.

**Comment by bluebill1049** [comment]
I would suggest breaking them into individual forms with each `useForm`.

**Reply by sabazusi** [reply]
OK, thank you.

**Reply by kondei** [reply]
@bluebill1049 
Sorry to interrupt.
I would like to know that this is intentional behavior of useFormState or not.

**Reply by bluebill1049** [reply]
hey, @kondei all the validation result is driven by the schema validation result. If something is not working, i would recommend doing a quick debug on the resolver 

```tsx
resolver: async (data, context, options) => {
  // you can debug your validation schema here
  console.log('formData', data)
  console.log('validation result', await anyResolver(schema)(data, context, options))
  return anyResolver(schema)(data, context, options)
},
```

**Reply by kondei** [reply]
I see.
Thank you.

**Reply by rhythm1019** [reply]
I am looking to debug in the manner shown here, however where is `anyResolver` defined, i.e. where should I import it from? It is not exported from the `react-hook-form` package.

**Reply by rhythm1019** [reply]
> I am looking to debug in the manner shown here, however where is `anyResolver` defined, i.e. where should I import it from? It is not exported from the `react-hook-form` package.

Ok, I understand what's being implied, that the `anyResolver` can be, for example, `yupResolver`.  There's no such thing called `anyResolver`.
